### DISCUSSION TITLE
[26.11.2024] Blog 7 - Architektur

### DISCUSSION THREAD

**Discussion by Kartoffeleintopf** [bunki1224/Softwareengineering-Projekt#50] [Show and tell]
# Entscheidung
Für unser Projekt haben wir uns für eine Servicebasierte Architektur entschieden. Um diese Entscheidung zu treffen sind wir diverse Qualitätsmerkmale durchgangen um eine informierte Entscheidung zu treffen. Der folgende Inhalt des Blogs beschäftigt sich mit den von uns betrachteten Szenarien und unsere daraus resultierenden Entscheidung für unseren Architekturstil.

# Qualitätsmerkmale und Szenarien
![Baum](https://github.com/user-attachments/assets/9cdda54f-e59c-400d-aa4b-131ad0010b11)

### Szenario 1
Fehler oder Bugs werden im Code gefunden. Diese sollten so bald wie möglich korregiert/behoben werden. Dafür muss die Architektur des Projektes das erlauben. Um nicht beabsichtigte Nebeneffekte zu vermeiden müssen die Komponenten des Projekts ein Maß an Isolation aufweisen. Somit fällt ein Stil wie der Monolithische raus.
### Szenario 5
Der Code verfügt über Kommentare welcher es Entwicklern erleichtert den Code zu warten.
Beispiele hierfür sind:
- Bezeichnungen für Abschnitte im Code
- Zusammenfassung von Funktionen
- sinnvoll benannte Variablen und Funktionen
- Formattierte Struktur für Lesbarkeit
### Szenario 2
Wir haben entschieden uns auf nur eine Datenbank zu limitieren.
Grund hierfür ist der Bedarf alles an einem Ort zu speichern.
Davon versprechen wir uns eine Reduktion des Verwaltungsaufwands, der vermutlich bei mehr als einer auftritt.
Durch diese Entscheidung kommt der Stil der Microservice Architektur, bei Verwendung von mehr als einem einzigen Service, nicht mehr in Frage.
### Szenario 3
Intuitives Erscheinungsbild. Aktuell besteht keine daraus resultierende Architektureinschränkung welche wir erkennen können.
### Szenario 4
Antwort auf Nutzereingabe binnen kurzer Zeit, dies soll einer responsiven Nutzererfahrung dienen.
Diese Anforderung muss durch den Architekturstil gegeben sein.
Durch diese Entscheidung erfolgt keine Einschränkung unserer Auswahl.

## Schlusswort

Hoffentlich könnt ihr nun unsere Entscheidung besser nachvollziehen.
Danke fürs Lesen!

> Euer TripAhead Team (CN)

**Comment by OmarMoujahed** [comment]
Ein sehr gut strukturierter Beitrag, der die Entscheidungskriterien und Qualitätsmerkmale übersichtlich darstellt. Besonders die Szenarien-Analyse bietet wertvolle Einblicke, wie verschiedene Anforderungen priorisiert und umgesetzt werden können. Die klare Aufteilung in Wartung, Benutzbarkeit und Support macht die Inhalte leicht nachvollziehbar. Danke für die fundierte Darstellung!

Mit freundlichen Grüßen
Team HÆLTH

**Comment by themsgcentre** [comment]
Ihr zeigt eine klare und durchdachte Herangehensweise an die Anforderungen und Herausforderungen. Besonders positiv fällt die strukturierte Planung und die Berücksichtigung von Kernaspekten wie Wartbarkeit und Benutzerfreundlichkeit auf. Die offenen Diskussionen und der Fokus auf teamübergreifende Kommunikation sind bemerkenswert und tragen sicherlich zur Qualität des Endprodukts bei. Weiter so – ein vielversprechender Ansatz!

Viele Grüße,
Projekt TDC (HM)

**Comment by schmid-di** [comment]
Ein äußerst interessanter Einblick in eure technische Entscheidungsfindung! Die Wahl einer servicebasierten Architektur für eure Reiseplanungs-App erscheint sehr durchdacht, insbesondere in Hinblick auf Skalierbarkeit und Wartbarkeit. Die Betrachtung verschiedener Szenarien, um eine fundierte Entscheidung zu treffen, ist ein starkes Beispiel für eine systematische Herangehensweise. Es ist spannend zu sehen, wie ihr zentrale Aspekte wie lose Kopplung und Modularität integriert, um eine flexible und zukunftssichere Lösung zu schaffen. Ein vielversprechender Ansatz, der das Potenzial eurer App weiter unterstreicht

Euer UMOC-Team
/JM /PL
